draft-ietf-quic-qpack.md

```diff
@@ -550,18 +550,36 @@ the block.  Blocking decoders use the Largest Reference to determine when it is
 safe to process the rest of the block.
 
 `Base Index` is used to resolve references in the dynamic table as described in
-{{indexing}}.  To save space, Base Index is encoded relative to Largest
-Reference using a one-bit sign flag.
+{{indexing}}.
 
-    baseIndex = largestReference + deltaBaseIndex
+To save space, Base Index is encoded as a relative to the Base Index using a
+one-bit sign and the `Delta Base Index` value.  A sign bit set to 0 indicates
+that the Base Index has a greater absolute index than the Largest Reference; the
+value of Delta Base Index is added to the Largest Reference to determine the
+absolute value of the Base Index.  A sign bit of 1 indicates that the Base Index
+is smaller than the Base Index.
 
-If the encoder inserted entries to the table while the encoding the block,
-Largest Reference will be greater than Base Index, so deltaBaseIndex will be
-negative and encoded with S=1.  If the block did not reference the most recent
-entry in the table and did not insert any new entries, Largest Reference will be
-less than Base Index, so deltaBaseIndex will be positive and encoded with S=0.
-When Largest Reference and Base Index are equal, deltaBaseIndex is 0 and encoded
-with S=0.
+That is, after turning a sign bit of 1 to -1 and 0 to 1, the absolute Base Index
+is:
+
+~~~
+    baseIndex = largestReference + sign * deltaBaseIndex
+~~~
+
+If the encoder inserted entries to the table while the encoding the header
+block, Largest Reference will be greater than Base Index, so the encoded
+difference is negative and the sign bit will be 1.  If the header block did not
+reference the most recent entry in the table and did not insert any new entries,
+Base Index will be greater than the Largest Reference, so the delta will be
+positive and the sign bit will be 0.
+
+When Largest Reference and Base Index are equal, the Delta Base Index is encoded
+with a zero sign bit.  A sign bit set to 1 when the Delta Base Index is 0 MUST
+be treated as a decoder error.
+
+A header block that does not reference the dynamic table can use any value for
+Base Index and Largest Reference; sending zero values for both fields is the
+most efficient.
```

Comment:
You can use any value for Base Index, and zero is the most efficient.  I'd argue that zero for Largest Reference is probably a SHOULD, given that it has a defined meaning and isn't just used for relative indexing.

Reply:
Yeah, I considered that, but it's not a SHOULD really, especially since we have defined Largest Reference already (it's the largest absolute index in the block).  I'll adjust and let you decide if you want to merge.

Reply:
Having no references and the value not being zero is a violation of what that field means.  More generally, I think a decoder that wanted to have an error case for "You didn't actually reference the Largest Reference" would be well within its rights, and we might want to either mention that and define the error code or explicitly prohibit that.

Alan's doc, his call.

Reply:
Rather than bog this PR down, let's fix that separately.